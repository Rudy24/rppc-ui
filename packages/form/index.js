/*
 * @Author: 宋绍华
 * @Date: 2021-10-19 09:44:55
 * @LastEditTime: 2021-10-19 10:04:24
 * @LastEditors: 宋绍华
 * @Description:
 * @FilePath: \rppcui\packages\table\index.js
 */
// 引入组件
import RpFrom from './form'

// 提供 install 安装方法，供按需引入
RpFrom.install = (Vue) => {
  // 注册组件
  Vue.component(RpFrom.name, RpFrom)
}
// 暴露组件
export default RpFrom
