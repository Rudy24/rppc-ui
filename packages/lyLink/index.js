// 引入组件
import lyLink from './src'

// 提供 install 安装方法，供按需引入
lyLink.install = (Vue) => {
  // 注册组件
  Vue.component(lyLink.name, lyLink)
}
// 暴露组件
export default lyLink
