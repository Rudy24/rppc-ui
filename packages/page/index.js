/*
 * @Author: 宋绍华
 * @Date: 2021-10-19 09:44:55
 * @LastEditTime: 2021-10-19 10:04:24
 * @LastEditors: 宋绍华
 * @Description:
 * @FilePath: \rppcui\packages\table\index.js
 */
// 引入组件
import RpPage from './page'

// 提供 install 安装方法，供按需引入
RpPage.install = (Vue) => {
  // 注册组件
  Vue.component(RpPage.name, RpPage)
}
// 暴露组件
export default RpPage
