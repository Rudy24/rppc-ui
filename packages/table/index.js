/*
 * @Author: 宋绍华
 * @Date: 2021-10-19 09:44:55
 * @LastEditTime: 2021-10-19 10:04:24
 * @LastEditors: 宋绍华
 * @Description:
 * @FilePath: \rppcui\packages\table\index.js
 */
// 引入组件
import RpTable from './src/table'

// 提供 install 安装方法，供按需引入
RpTable.install = (Vue) => {
  // 注册组件
  Vue.component(RpTable.name, RpTable)
}
// 暴露组件
export default RpTable
