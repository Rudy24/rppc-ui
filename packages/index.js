/*
 * @Author: 宋绍华
 * @Date: 2021-10-18 17:32:17
 * @LastEditTime: 2021-10-19 15:56:08
 * @LastEditors: 宋绍华
 * @Description:
 * @FilePath: \rppcui\packages\index.js
 */
// 引入组件
// eslint-disable-next-line object-curly-newline
import { Table, TableColumn, Pagination, Form, FormItem, Input, Loading } from 'element-ui'
import Vue from 'vue'
import 'element-ui/lib/theme-chalk/index.css'
import lyLink from './lyLink'
import RpTable from './table'
import RpPage from './page'
import RpForm from './form'

// 引入使用的element组件
const eleComps = [Table, TableColumn, Pagination, Form, FormItem, Input, Loading]
eleComps.forEach((item) => {
  Vue.use(item)
})

// 存放组件的数组
const components = [lyLink, RpTable, RpPage, RpForm]

// 定义 install 方法，接收 Vue 作为参数。
const install = (V) => {
  // 判断是否安装
  if (install.installed) return
  // 遍历 components 数组，来进行全局注册
  components.map((component) => V.component(component.name, component))
}

// 判断是否是直接引入文件
if (typeof window !== 'undefined' && window.Vue) {
  install(window.Vue)
}

export default install

export {
  // 导出的对象必须具有 install，才能被 Vue.use() 方法安装
  install,
  lyLink,
  RpTable,
  RpPage,
  RpForm
}
