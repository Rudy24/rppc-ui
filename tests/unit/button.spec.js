/*
 * @Author: 宋绍华
 * @Date: 2021-10-18 20:32:45
 * @LastEditTime: 2021-10-21 14:25:42
 * @LastEditors: 宋绍华
 * @Description:
 * @FilePath: \rppcui\tests\unit\button.spec.js
 */
import { mount } from '@vue/test-utils'
import { lyLink } from '../../packages'

test('test lyLink link', () => {
  const wrapper = mount(lyLink, {
    propsData: {
      href: 'www.baidu.com',
      type: 'primary1'
    },
    slots: {
      default: 'button text'
    }
  })
  expect(wrapper.props().type).toContain('primary')
  expect(wrapper.props().href).toEqual('www.baidu.com')
  expect(wrapper.text()).toBe('button text')
  expect(wrapper.html()).toMatchSnapshot()
  // wrapper.unmount();
})
