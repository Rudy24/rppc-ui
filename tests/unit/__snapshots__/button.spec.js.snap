// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test lyLink link 1`] = `<a href="www.baidu.com" class="ly-link-primary1">button text</a>`;
